dart/lib/src/protocol/sentry_span.dart

```diff
@@ -44,14 +46,23 @@ class SentrySpan extends ISentrySpan {
     if (status != null) {
       _status = status;
     }
-    _timestamp = endTimestamp ?? getUtcDateTime();
+
+    if (endTimestamp?.isBefore(_startTimestamp) ?? false) {
+      _hub.options.logger(
+        SentryLevel.warning,
+        'End timestamp ($endTimestamp) cannot be before start timestamp ($_startTimestamp)',
+      );
+      _endTimestamp = getUtcDateTime();
```

Comment:
This is a good idea, but `getUtcDateTime()` can still be technically before the given `_startTimestamp`.
This needs to be tried/tested and see if Sentry.io throws an ingestion error or just an UI warning, otherwise we have to fix on the SDK side.

Reply:
AFAIU, Sentry JS simply [discards spans if end timestamp is before start timestamp](https://github.com/getsentry/sentry-javascript/blob/889024a8b9da0664447f157eacaaf8713d6cc7d4/packages/tracing/src/idletransaction.ts#L131). 
Should this approach be applicable here?

```dart
      this.spanRecorder.spans = this.spanRecorder.spans.filter((span: Span) => {
        // If we are dealing with the transaction itself, we just return it
        if (span.spanId === this.spanId) {
          return true;
        }
         ...
        const keepSpan = span.startTimestamp < endTimestamp;
        if (!keepSpan) {
          logger.log(
            '[Tracing] discarding Span since it happened after Transaction was finished',
            JSON.stringify(span, undefined, 2),
          );
        }
        return keepSpan;
      });

```

Reply:
Yes, otherwise it'll be discarded in the event ingestion, so no need to send the data over.

Reply:
I can't find a suitable place to filter invalid spans out. 
Possible places: 
1. in Hub.captureTransaction, before [item.client.captureTransaction(transaction)](https://github.com/getsentry/sentry-dart/blob/c25be502cb39fcc5699bdf241c20da4f14856519/dart/lib/src/hub.dart#L463): can't modify transaction
2. in SentryTracer.finish, before [_hub.captureTransaction(transaction)](https://github.com/getsentry/sentry-dart/blob/c25be502cb39fcc5699bdf241c20da4f14856519/dart/lib/src/sentry_tracer.dart#L95): can't modify transaction
3. in constructor of SentryTransaction, while initializing spans: [spans = _tracer.children](https://github.com/getsentry/sentry-dart/blob/c25be502cb39fcc5699bdf241c20da4f14856519/dart/lib/src/protocol/sentry_transaction.dart#L55).where(...): 
#(3) might work, but smells. Any recommendation?

Reply:
@fatihergin I believe `SentryTracer#finish` is the way to go, with the for loop from L67-L73, just remove the child span from the `_children` list if the timestamp is out of bounds.